diff --git a/Dockerfile b/Dockerfile
index 5a33fe4..de563d1 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -59,6 +59,7 @@ RUN apt-get update && apt-get install -y \
 	s3cmd=1.1.0* \
 	ubuntu-zfs \
 	libzfs-dev \
+	golang \
 	--no-install-recommends
 
 # Get lvm2 source for compiling statically
@@ -82,20 +83,23 @@ RUN cd /usr/src/lxc \
 	&& make install \
 	&& ldconfig
 
+RUN mkdir /usr/local/go1.4 && curl -fsSL https://golang.org/dl/go1.4.3.linux-amd64.tar.gz | \
+    tar -v -C /usr/local/go1.4 -xz --strip 1
+ENV GOROOT_BOOTSTRAP /usr/local/go1.4
+
+
 # Install Go
-ENV GO_VERSION 1.4.3
+ENV GO_VERSION 1.5.1
 RUN curl -sSL https://golang.org/dl/go${GO_VERSION}.src.tar.gz | tar -v -C /usr/local -xz \
 	&& mkdir -p /go/bin
 ENV PATH /go/bin:/usr/local/go/bin:$PATH
 ENV GOPATH /go:/go/src/github.com/docker/docker/vendor
 RUN cd /usr/local/go/src && ./make.bash --no-clean 2>&1
 
